benchmarks/neutron/Snakefile

```diff
@@ -0,0 +1,58 @@
+rule neutron_generate:
+	input:
+                script="benchmarks/neutron/analysis/gen_particles.cxx",
+	params:
+		NEVENTS_GEN=1000,
+		th_max=5.7,
+		th_min=2.0
+	output:
+		GEN_FILE="sim_output/neutron/neutron_{P}GeV.hepmc"
+	shell:
+		"""
+mkdir -p sim_output/neutron
+root -l -b -q '{input.script}({params.NEVENTS_GEN},"{output.GEN_FILE}", "neutron", {params.th_min}, {params.th_max}, 0., 360., {wildcards.P})'
```

Comment:
It would make sense to use a single particle neutron simulations steering files from campaigns, like in tracking_performance benchmark. That way we can run this benchmark over campaigns.

Reply:
Thanks for the suggestion.  Is it possible to have the theta used in the steering file be relative to the proton direction?   Since this benchmark is for the insert, which is offset to the right by about 10 cm to accommodate the outgoing hadron beampipe it would make the most sense to have this be relative to the proton direction.  If the steering file has a feature where we can rotate to a different coordinate system, I'll take your suggestion; otherwise I will keep this as it is

Reply:
If i did what you are proposing, and I increase the range in theta (which would be about the electron axis instead of the proton axis) such that it encompasses the detector, only about 1/3 to 1/2 of the events would have the neutron hitting the insert.  Some of those neutrons would hit the endcap and would not be useful in this analysis, so we'd just be making the benchmark take longer to run or reducing our statistics.  

Here's what I propose:

1. Merge this pull request as it is
2. Do a pull request in DD4hep with a modification allowing steering files to define theta relative to an axis other than the z axis,
3. Once that is complete, we can modify this benchmark to use the steering files with the theta relative to the proton axis.  

What do you think?

Reply:
I don't think we need the complication for DD4hep, as getting rid of hepmc generator is not the goal here. If the hit to statistics is bad (i.e. significantly less than 50% reduction), then we probably can't do it. But if it hits endcap just a reasonable amount, then that sample can be (re-)used for both detectors and for studies of performance in overlaps.

Reply:
ok, I can include this feature in a follow-up pull request.